packages/synapse-react-client/src/components/EntityLink.tsx

```diff
@@ -38,14 +38,24 @@ export const EntityLink = (props: EntityLinkProps) => {
     entityId = entityOrId
   }
 
-  const { data: fetchedEntity, error } = useGetEntityHeader(
+  const fetchHeader = !!entityId && typeof entityOrId === 'string'
+
+  const { data: fetchedEntity, isLoading } = useGetEntityHeader(
     entityId,
     versionNumber,
     {
-      enabled: !!entityId && typeof entityOrId === 'string',
+      enabled: fetchHeader,
     },
   )
 
+  // We don't get 4XX errors from useGetEntityHeader, so fetch the entity object to get an error response
+  // See PLFM-7989
+  const fetchFullEntity =
+    fetchHeader && fetchedEntity == undefined && !isLoading
+  const { error } = useGetEntity(entityId, versionNumber, {
+    enabled: fetchFullEntity,
+  })
```

Comment:
Fix a case where we lost the error status by changing the implementation of `useGetEntityHeader`. This bug causes us to show a skeleton loader instead of a red/yellow error chip component.